managed-pointer/cpp-20/D0567R1.md

```diff
@@ -5,222 +5,384 @@
+
+If any memory allocation or deallocation operations fail then an exception is thrown and stored in the future object that is returned by the operations which triggered the allocation operation.
+
+#### 3.3.5. Synchronisation
+
+For an **execution context** to be the accessible **execution context**, if it is not already, a synchronisation operation is required. A synchronisation operation is an implementation defined asynchronous st of commands which moves the data from the currently accessible memory region to another memory region which returns a future object that can be used to wait for the operation to complete. From the point at which a synchronisation operation is triggered the currently accessible memory region is no longer accessible. Once the synchronisation point is complete (i.e. the future returned from the synchronisation operation is complete) the memory region the data is being synchronised to is now the accessible memory region.
+
+Synchronisation operations are course grained synchronisation in that they synchronise the entire managed memory allocation of a `managed_ptr`.
+
+> This limited may be relaxed in a future version of the paper when a `managed_ptr` can be concurrently accessed in more than one memory region. In this case there sychronisation could be performed at smaller granularities using mapping operations or atomic operations. This is further described in section 5.1.
+
+There are three sycnhronisation operations; `get`, `put` and `sync`, each taking one or more `managed_ptr` parameters. These sycnhronisation operations perform the following sycnhronisation for each `managed_ptr` parameter:
+
+* The `get` operation synchronises the data at the memory region at a particular *execution context* to the memory region at the *execution context* associated with the `managed_ptr`.
+* The `put` operation synchronises the data at the memory region at the *execution context* associated with the `managed_ptr` to the memory region at a particular *execution context*.
+* The `sync` operation synchronises the data at the memory region at one particular *execution context* to the memory region at another particular *execution context*.
+
+> In the previous revision of this paper it was also possible to implicitly trigger sycnhornisation operations by simplly passing a `managed_ptr` to a control structure such as `async`. However this has been removed due to a limitation that may effect implementations. This is described in section 5.2.
+
+There are a further three sycnhronisation operations; `then_get`, `then_put` and `then_sync`, which are equivelants of `get`, `put` and `sync` but take a predecate future parameter that the operation must wait on before executing. These are used to provide a continuations interface for sycnhronisation operations. Equivelant operations to these are also required as member functions on the future type, this is described further in section 3.4.
```

Comment:
take a predecate -> take a predicate? 
synchronisation -> synchronization
Equivelant -> equivalent
"type, this is" -> type. This is"